lib/samly/provider.ex

```diff
@@ -57,6 +57,16 @@ defmodule Samly.Provider do
 
     Application.put_env(:samly, :idp_id_from, idp_id_from)
 
+    refresh_providers()
+  end
+
+  @doc """
+  Refresh the provider configuration, allowing runtime-configuration to be applied after
+  application start.
+  """
+  def refresh_providers do
+    opts = Application.get_env(:samly, Samly.Provider, [])
```

Comment:
Wouldn't you need to refresh everything below line 39 also?

Reply:
The only field loaded there is `idp_id_from` - that's a global setting, not provider specific, so I kind of assumed that wouldn't change at runtime. Easy enough to move if you think there's a case where it might, though.